nncf/common/graph/graph.py

```diff
@@ -33,51 +33,73 @@ class NNCFNode:
     Class describing nodes used in NNCFGraph.
     """
 
-    def __init__(self, node_id: int, node_name: NNCFNodeName, data: dict = None):
-        self.node_id = node_id
-        self.data = data if data else {}
-        self.data[NNCFGraph.NODE_NAME_ATTR] = node_name
+    ID_NODE_ATTR = "id"
+    NODE_NAME_ATTR = "node_name"
+    KEY_NODE_ATTR = "key"
+    NODE_TYPE_ATTR = "type"
+    METATYPE_ATTR = "metatype"
+    LAYER_NAME_ATTR = "layer_name"
+    LAYER_ATTRIBUTES = "layer_attributes"
+    IGNORED_ALGOS_ATTR = "ignored_algos"
+    IS_IN_ITERATION_SCOPE_NODE_ATTR = "is_in_iteration_scope"
+    IS_INTEGER_INPUT_NODE_ATTR = "is_integer_input"
+    IS_SHARED_ATTR = "is_shared"
+
+    def __init__(self, attributes: Dict[str, Any]):
+        self._attributes = attributes
+
+    @property
+    def attributes(self) -> Dict[str, Any]:
+        return self._attributes
+
+    @property
+    def node_id(self) -> int:
+        return self._attributes[NNCFNode.ID_NODE_ATTR]
+
+    @property
+    def node_key(self) -> str:
+        return self._attributes[NNCFNode.KEY_NODE_ATTR]
 
     @property
     def node_name(self) -> NNCFNodeName:
-        return self.data.get(NNCFGraph.NODE_NAME_ATTR)
+        return self._attributes[NNCFNode.NODE_NAME_ATTR]
 
     @property
     def metatype(self) -> Type[OperatorMetatype]:
-        return self.data.get(NNCFGraph.METATYPE_ATTR)
+        return self._attributes[NNCFNode.METATYPE_ATTR]
 
     @property
     def node_type(self) -> str:
-        return self.data.get(NNCFGraph.NODE_TYPE_ATTR)
+        return self._attributes[NNCFNode.NODE_TYPE_ATTR]
 
     @property
-    def layer_name(self) -> LayerName:
-        return self.data.get(NNCFGraph.LAYER_NAME_ATTR)
+    def layer_name(self) -> Optional[LayerName]:
```

Comment:
Is it really need here type aliases `LayerName` and `NNCFNodeName`. How about use `str` instead of create extra 2 names of `str` type?

Reply:
@vshampor, could you answer of this comment, because you introduced `LayerName`?

Reply:
The two entities are different in purpose and must have different types. That both types can both currently be sufficiently represented by `str` is a coincidence and does not bring any overhead at all.

Reply:
But now it's `str`. In this logic we can create own aliases for each argument, but it's looks like create aliases to create alias. And this can confuse when you expect `LayerName` class but got `str`. 

I think that we should avoid creation nonfunctional entities. 
To describe different between `LayerName` and `NNCFNodeName` better use name of functions and arguments and docstring.